pkg/controller/factory/controller_context.go

```diff
@@ -96,19 +96,17 @@ func (c syncContext) enqueueKeys(keys ...string) {
 // (or its tombstone) is a namespace  and it matches a name of any namespaces
 // that we are interested in
 func namespaceChecker(interestingNamespaces []string) func(obj interface{}) bool {
-	interestingNamespacesSet := sets.New(interestingNamespaces...)
-
 	return func(obj interface{}) bool {
 		ns, ok := obj.(*corev1.Namespace)
 		if ok {
-			return interestingNamespacesSet.Has(ns.Name)
+			return slices.Contains(interestingNamespaces, ns.Name)
 		}
 
 		// the object might be getting deleted
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if ok {
 			if ns, ok := tombstone.Obj.(*corev1.Namespace); ok {
-				return interestingNamespacesSet.Has(ns.Name)
+				return slices.Contains(interestingNamespaces, ns.Name)
```

Comment:
I'm in favor of using `slices.Contains` instead of constructs like:

```
sets.New(HostnameMatchSpecCandidates(hostname)...).Has(matchSpec)
```

But I'm not sure about cases like this one, as we're going from O(1) to O(n) and I'm not sure how often these are called.

Reply:
I’ve restored two uses of sets where performance might matter.